### PR TITLE
deploy: Adds support for other container engines (ie. podman)

### PR DESCRIPTION
**Pull request by ngharo**
## Proposed changes
Adds `--container-engine` command line option.
Can also be set via `container_engine` config option. Used underscore for possible future env-var friendliness.

The main driver for this change is to add support for `podman`.

This also changes container image references to fully qualified, pointed at docker.io registry where runiac images are published. 

## Issues for these changes
Users may have unexpected results if they're using any container engine other than default docker.

## Types of changes
- Changed any container image references to fully qualified, pointed at docker.io registry where runiac images are published. 

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor (changes to code, which do not change application behavior)

## Further comments
Tested using podman version 3.0.1

### CODE REVIEW

**Review by tiny-dancer — CHANGES_REQUESTED** [review]
Let's add 'container engine' support to the config file as well.

**Comment by ngharo** [comment]
I was wanting that too. I need to dig more into how Viper and Cobra are working together.

I should be able to figure it out quickly and update this PR.

**Comment by tiny-dancer** [comment]
@ngharo check out `func getDockerfileForBuild() string {`

In `cmd/cli/deploy` for a reference. There may be a better overall connection between the two but that's the existing precedent example for config not passed into the runiac container

(On mobile app sorry for formatting and lack of deep link)

**Comment by ngharo** [comment]
Thanks for the pointer. Way easier than I thought.

File: cmd/cli/cmd/deploy.go [hunk 1]
```diff
@@ -218,16 +222,24 @@ func appendE(slice []string, arg string, val string) []string {
...
+		fmt.Printf("please add '%s' to the path\n", getContainerEngineForBuild())
 	}
 }
 
 func checkInitialized() bool {
 	return InitAction()
 }
 
+func getContainerEngineForBuild() string {
+	if viper.GetString("container_engine") != "" {
```
Comment (tiny-dancer) [inline cmd/cli/cmd/deploy.go:236]: Flip flop these returns, for consistency we want to respect cli arg present over declarative config (yaml) present.

**Review by tiny-dancer — CHANGES_REQUESTED** [review]
Looks good, one minor requested change and then an optional change to consolidate the multiple function calls to one top level function  variable.

**Comment by ngharo** [comment]
What do you think of this approach?

The exported variable `ContainerEngine` gets set to its default value where it's declared.
The var may be updated by passing the command line arg.
If no cmd line arg, check the config file.

I like this because when I first started looking at this code I was wondering: Why aren't Cobra default value arguments being set in flags and instead defaults asserted via a function call. I get it now that it's for setting command line precedence but it was still little confusing to me.

**Comment by tiny-dancer** [comment]
@ngharo looks great, tidy up the tests failing and we’ll be good to go.

**Comment by ngharo** [comment]
Yeah, about that 🤣 

Since it's using Cobra to know if the command option is set, I tried mocking the command line. I can't get viper.Set() in tests to be seen in viper.GetString() in deploy.go.

Pushed up WIP commit. Will revisit this soon.

**Comment by ngharo** [comment]
It was non-trivial to reset `cobra.Flags().Changed()` state in tests. A simple solution was to change the order of the tests, so admittedly, it's a little cheesed. But this only matters if testing things using `cobra.Flags().Changed()` logic, which I can't imagine will come up for anything but option precedence tests, which is covered here.

**Review by tiny-dancer — APPROVED** [review]

**Comment by tiny-dancer** [comment]
Thanks for the bump, LGTM

**Pull request merged**